svc/pkg/game/ops/namespace-get/src/lib.rs

```diff
@@ -34,20 +34,32 @@ async fn handle(
 		.map(common::Uuid::as_uuid)
 		.collect::<Vec<_>>();
 
-	let namespaces = sql_fetch_all!(
-		[ctx, Namespace]
-		"
-		SELECT namespace_id, game_id, create_ts, display_name, version_id, name_id
-		FROM db_game.game_namespaces
-		WHERE namespace_id = ANY($1)
-		ORDER BY display_name
-		",
-		&namespace_ids,
-	)
-	.await?
-	.into_iter()
-	.map(Into::<backend::game::Namespace>::into)
-	.collect::<Vec<_>>();
+	let namespaces = ctx
+		.cache()
+		.immutable()
+		.fetch_all_proto("namespace", namespace_ids, |mut cache, namespace_ids| {
+			let ctx = ctx.base();
+			async move {
+				let namespaces = sql_fetch_all!(
+					[ctx, Namespace]
+					"
+					SELECT namespace_id, game_id, create_ts, display_name, version_id, name_id
+					FROM db_game.game_namespaces
+					WHERE namespace_id = ANY($1)
+					ORDER BY display_name
+					",
+					&namespace_ids,
+				)
+				.await?;
+
+				for namespace in namespaces {
+					cache.resolve(&namespace.namespace_id.clone(), namespace.into());
+				}
+
+				Ok(cache)
+			}
```

Comment:
This might be a bit pedantic, but this can be shortened a little into a full iterator statement. Though as the code is right now, it is a bit easier to see what's going on, as in getting namespaces, then iterating over each. Would be curious to hear your thoughts 😀 

```suggestion
			async move {
				sql_fetch_all!(
					[ctx, GameUser]
					"
					SELECT gu.game_user_id, gu.user_id, gu.token_session_id, gu.namespace_id, gu.create_ts, l.link_id, gu.deleted_ts
					FROM db_game_user.game_users AS gu
					LEFT JOIN db_game_user.links AS l ON l.new_game_user_id = gu.game_user_id
					WHERE gu.game_user_id = ANY($1)
					",
					game_user_ids,
				)
				.await?
				.into_iter()
				.for_each(|game_user| {
					cache.resolve(
						&game_user.game_user_id.clone(),
						game_user.into(),
					);
				});

				Ok(cache)
			}
```